packages/runtime/container-runtime/src/opLifecycle/opDecompressor.ts

```diff
@@ -23,12 +23,7 @@ export class OpDecompressor {
     private processedCount = 0;
 
     public processMessage(message: ISequencedDocumentMessage): IMessageProcessingResult {
-        // We're checking for compression = true or top level compression property so
-        // that we can enable compression without waiting on all ordering services
-        // to pick up protocol change. Eventually only the top level property should
-        // be used.
-        if (message.metadata?.batch === true
-            && (message.metadata?.compressed || message.compression !== undefined)) {
+        if (message.metadata?.batch === true && message.compression !== undefined) {
```

Comment:
We should check for ```message.compression === CompressionAlgorithms.lz4``` as well here to make sure the batch is a supported compression algorithm (same check as is in the single compressed message path). I don't think there exists any logic to explicitly fail if a compressed message comes through with an unsupported compression scheme. It might be worth adding an explicit failure here with a useful error message now that we're making more progress on this work.